lib/internal/README.md

```diff
@@ -1,6 +1,13 @@
 # Internal Modules
 
-The modules in `lib/internal` are intended for internal use in Node.js core
-only, and are not accessible with `require()` from user modules. These modules
-can be changed at **any** time. Reliance on these modules outside of core
-is **not supported** in any way.
+The modules located in `lib/internal` directory are exclusively meant
+for internal usage within the Node.js core. They are not intended to
+be accessed via user modules `require()`. These modules may change at
+any point in time. Relying on these internal modules outside the core
+is not supported and can lead to unpredictable behavior.
+
+In certain scenarios, accessing these internal modules for debugging or
+experimental purposes might be necessary. Node.js provides the `--expose-internals`
+flag to expose these modules to userland code. Exercise caution when using
+this flag, as it can lead to unexpected results and is primarily intended
+for advanced users and debugging purposes.
```

Comment:
We've always been a bit careful about documenting this and I think we need to be careful about saying things like "expose these modules to userland code" as we do not want to encourage the ecosystem to use this. We should explicitly state that this flag is *ONLY* supported for use in Node.js' own tests and benchmarks

Reply:
The next line states to use caution, if that's not good enough, I can make it more explicitly unstable.

Reply:
I'd prefer we flat out said "Don't Use This", "It is not supported", etc